src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpHandler.cs

```diff
@@ -1632,67 +1108,36 @@ private void SetRequestHandleTlsOptions(SafeWinHttpHandle requestHandle)
 
         private void SetRequestHandleClientCertificateOptions(SafeWinHttpHandle requestHandle, Uri requestUri)
         {
-            // Must be HTTPS scheme to use client certificates.
-            if (requestUri.Scheme != UriSchemeHttps)
+            if (requestUri.Scheme != WinHttpRequestState.UriSchemeHttps)
             {
                 return;
             }
 
-            // Get candidate list for client certificates.
-            X509Certificate2Collection certs;
+            X509Certificate2 clientCertificate = null;
             if (_clientCertificateOption == ClientCertificateOption.Manual)
             {
-                certs = ClientCertificates;
+                clientCertificate = WinHttpCertificateHelper.GetEligibleClientCertificate(ClientCertificates);
```

Comment:
Is this behavior similar to Desktop? (Checking certificates supplied by the user against the policy?)
I'm thinking of certain situations where non-standard servers are used that may be accepting certs missing the Client OID.